bokehjs/src/coffee/ar/transforms.coffee

```diff
@@ -0,0 +1,113 @@
+HasProperties = require "../common/has_properties"
+
+class Transform extends HasProperties
+
+class AutoEncode extends Transform
+  type: "AutoEncode"
```

Comment:
This can be further simplified:

```coffeescript
generateTransform = (name) ->
  # Create a holder object since we need something to take the dynamic assign
  holder = {}
  holder[name] = class extends Transform
    type: name
  return holder[name]

AutoEncode = generateTransform "AutoEncode"
BinarySegment = generateTransform "BinarySegment"
```

I [created a pen][1] showing this code.

[1]: http://codepen.io/tswicegood/pen/PwgmEW?editors=101